generatebundlefile/ecr.go

```diff
@@ -187,12 +327,130 @@ func removeStringSlice(l []string, item string) []string {
-// func remove[T comparable](l []T, item T) []T {
-// 	for i, other := range l {
-// 		if other == item {
-// 			return append(l[:i], l[i+1:]...)
-// 		}
-// 	}
-// 	return l
-// }
+// GetPublicAuthToken gets an authorization token from ECR public
+func (c *ecrPublicClient) GetPublicAuthToken() (string, error) {
+	authTokenOutput, err := c.GetAuthorizationToken(context.TODO(), &ecrpublic.GetAuthorizationTokenInput{})
+	if err != nil {
+		return "", errors.Cause(err)
+	}
+	authToken := *authTokenOutput.AuthorizationData.AuthorizationToken
+
+	return authToken, nil
+}
+
+// GetPublicAuthConfig formats the Auth token to skopeo format, and adds to the ECR Public client struct
+func (c *ecrPublicClient) GetPublicAuthConfig() (*docker.AuthConfiguration, error) {
+	authToken, err := c.GetPublicAuthToken()
+	if err != nil {
+		return nil, errors.Cause(err)
+	}
+	// Decode authorization token to get credential pair
+	creds, err := base64.StdEncoding.DecodeString(authToken)
+	if err != nil {
+		return nil, errors.Cause(err)
+	}
+	// Get password from credential pair
+	credsSplit := strings.Split(string(creds), ":")
+	password := credsSplit[1]
+
+	// Construct docker auth configuration
+	authConfig := &docker.AuthConfiguration{
+		Username: "AWS",
+		Password: password,
+	}
+
+	return authConfig, nil
+}
+
+// GetAuthToken gets an authorization token from ECR
+func (c *ecrClient) GetAuthToken() (string, error) {
+	authTokenOutput, err := c.GetAuthorizationToken(context.TODO(), &ecr.GetAuthorizationTokenInput{})
+	if err != nil {
+		return "", errors.Cause(err)
+	}
+	authToken := *authTokenOutput.AuthorizationData[0].AuthorizationToken
+	return authToken, nil
+}
+
+// GetAuthConfig formats the Auth token to skopeo format, and adds to the ECR client struct
+func (c *ecrClient) GetAuthConfig() (*docker.AuthConfiguration, error) {
+	// Get ECR authorization token
+	authToken, err := c.GetAuthToken()
+	if err != nil {
+		return nil, errors.Cause(err)
+	}
+
+	// Decode authorization token to get credential pair
+	creds, err := base64.StdEncoding.DecodeString(authToken)
+	if err != nil {
+		return nil, errors.Cause(err)
+	}
+
+	// Get password from credential pair
+	credsSplit := strings.Split(string(creds), ":")
+	password := credsSplit[1]
+
+	// Construct docker auth configuration
+	authConfig := &docker.AuthConfiguration{
+		Username: "AWS",
+		Password: password,
+	}
+	return authConfig, nil
+}
+
+// copyImagePrivPubSameAcct will copy an OCI artifact from ECR us-west-2 to ECR Public within the same account.
+func copyImagePrivPubSameAcct(stsClient *stsClient, ecr *ecrClient, ecrPublic *ecrPublicClient, image Image) error {
+	source := fmt.Sprintf("docker://%s.dkr.ecr.us-west-2.amazonaws.com/%s:%s", stsClient.AccountID, image.Repository, image.Digest)
+	destination := fmt.Sprintf("docker://%s/%s:%s", ecrPublic.SourceRegistry, image.Repository, image.Digest)
+	fmt.Printf("Promoting %s\n to %s\n", source, destination)
```

Comment:
Use some sort of logging interface, or remove this.

Reply:
Added logging interface.